server/routes/api/user.js

```diff
@@ -1,7 +1,47 @@
 const express = require('express');
-
 const router = express.Router();
+const bcrypt = require('bcrypt');
+
+// User Model
+const User = require('../../models/user');
+
+// Test User route
+router.get('/test', (req, res) => res.json({ msg: 'User route works!' }));
+
+// POST User Registration
+router.post('/register', (req, res) => {
+  const email = req.body.email.trim();
```

Comment:
Email will never _need_ spaces in or around it, so I'd just toss the property `trim: true` in the model.

Reply:
I actually had it there, but changed.  

When a User is created the email address will be trimmed and saved.  If a new user is registering and enters an email that already exists, but with a space before/after I don't think it will match the trimmed email in the db, so would duplicate.

If we trim first we shouldn't have that issue.  I could be wrong on how the email would be compared though.

Reply:
That feels cosmetic for the backend. In my opinion, the backend should care about sanitizing input for security, but it should receive the _correct_ input in order to do what it needs to do. We can have the FE trim that. I would lean on the model property for that. Keep in mind our email validator _should_ catch that oddity of weird spacing so there isn't a duplicate.

Todo: I would test that. Try and make a request to your route (with the trims removed from the route and added to the model), and see if you get a duplicate.